content/3_Silver/Flood_Fill.mdx

```diff
@@ -372,117 +466,175 @@ static void floodfill(int r, int c, int color){
-	public static void ff(int x, int y){
-		Stack<Pair> s = new Stack<Pair>();
-		s.push(new Pair(x, y));
-		while(!s.isEmpty()){
-			Pair p = s.pop();
-			x = p.a; y = p.b;
-			if (x < 0 || x >= N || y < 0 || y >= M || g[x][y] == '#' || v[x][y])
+public class RoomCount {
+	public static final int R_CHANGE[] = {0, 1, 0, -1};
+	public static final int C_CHANGE[] = {1, 0, -1, 0};
+
+	private static boolean[][] visited;
+	private static char[][] building;
+	private static int rowNum;
+	private static int colNum;
+
+	public static void floodfill(int r, int c) {
+		// Note: you can also use an ArrayDeque for a BFS-based approach
+		Stack<int[]> frontier = new Stack<>();
+		frontier.push(new int[] {r, c});
+		while(!frontier.isEmpty()){
+			int[] p = frontier.pop();
+			r = p[0];
+			c = p[1];
+
+			if (r < 0 || r >= rowNum || c < 0 || c >= colNum
+					|| building[r][c] == '#' || visited[r][c])
 				continue;
-			v[x][y] = true;
-			for (int i = 0; i < 4; i++){
-				int nx = x+xd[i]; int ny = y+yd[i];
-				s.add(new Pair(nx,ny));
+
+			visited[r][c] = true;
+			for (int i = 0; i < 4; i++) {
+				frontier.add(new int[] {r + R_CHANGE[i], c + C_CHANGE[i]});
 			}
 		}
 	}
-	public static void main(String[] args) throws IOException
-	{
-		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
-		PrintWriter out = new PrintWriter(System.out);
-		StringTokenizer st = new StringTokenizer(br.readLine());
-		N = Integer.parseInt(st.nextToken());
-		M = Integer.parseInt(st.nextToken());
-		g = new char[N][M];
-		for (int i = 0; i < N; i++){
-			String str = br.readLine();
-			for (int j = 0; j < M; j++){
-				g[i][j] = str.charAt(j);
+
+	public static void main(String[] args) throws IOException {
+		BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer dims = new StringTokenizer(read.readLine());
+		rowNum = Integer.parseInt(dims.nextToken());
+		colNum = Integer.parseInt(dims.nextToken());
+
+		building = new char[rowNum][colNum];
+		for (int r = 0; r < rowNum; r++){
+			String row = read.readLine();
+			for (int j = 0; j < colNum; j++){
+				building[r][j] = row.charAt(j);
 			}
 		}
-		v = new boolean[N][M];
-		int ans = 0;
-		for (int i = 0; i < N; i++){
-			for (int j = 0; j < M; j++){
-				if(g[i][j] == '.' && !v[i][j]) {
-					ff(i,j);
-					ans++;
+
+		visited = new boolean[rowNum][colNum];
+		int roomNum = 0;
+		for (int i = 0; i < rowNum; i++){
+			for (int j = 0; j < colNum; j++){
+				if (building[i][j] == '.' && !visited[i][j]) {
+					floodfill(i, j);
+					roomNum++;
 				}
 			}
 		}
-		out.println(ans);
-		out.close();
+		System.out.println(roomNum);
 	}
 }
 ```
 
 </JavaSection>
 
+<PySection>
+
+```py
+row_num, col_num = [int(i) for i in input().split()]
+grid = [input() for _ in range(row_num)]
+visited = [[False for _ in range(col_num)] for _ in range(row_num)]
+
+
+def floodfill(r: int, c: int):
+	global visited
+
+	# Note: you can also use collections.deque for a BFS-based approach
```

Comment:
Either remove this note or elaborate on it (e.g., pop from beginning of the deque, push to end) otherwise this comment hangs here awkwardly.